woocommerce/woocommerce
# Add support for the Cart & Checkout in WooCommerce's declare compatibility feature

### All Submissions:






### Changes proposed in this Pull Request:



For a more detailed description, please read the one in this ticket #36413!

In summary, we added the Cart & Checkout Blocks to the list of features so that other extension developers might declare (in)compatibility with it. For the time being, we are preventing the display of the Cart & Checkout Blocks feature's UI settings. We may choose otherwise in the future, but for the time being, we only need the Cart & Checkout Blocks to be included in the list of WooCommerce features.

Closes #36413.






### How to test the changes in this Pull Request:



Checkout to this branch, then:

1. Create a new plugin or clone an existing one, install it, and activate it on your test website
2. Declare its compatibility with the Cart & Checkout Blocks by adding the following snippet into the main plugin file:

```PHP

add_action( 'before_woocommerce_init', function() {
	if ( class_exists( \Automattic\WooCommerce\Utilities\FeaturesUtil::class ) ) {
		\Automattic\WooCommerce\Utilities\FeaturesUtil::declare_compatibility( 'cart_checkout_blocks', __FILE__, true );
	}
} );

```

3. Clone the [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-blocks) plugin, install it, and activate it in your test website.
4. Check if we can get the list of compatible plugins with the Cart & Checkout Blocks with the WooCommerce Blocks plugin, and that the name of the plugin from step 1 is included. We can use the function `get_compatible_plugins_for_feature` from the `FeaturesController` API. For example, we can add the following snippet within the main plugin file:

```PHP

add_action(
	'woocommerce_init',
	function() {
		if ( class_exists( '\Automattic\WooCommerce\Utilities\FeaturesUtil' ) ) {
			var_dump(
				\Automattic\WooCommerce\Utilities\FeaturesUtil::get_compatible_plugins_for_feature( 'cart_checkout_blocks' )
			);
		}
	}
);

```



### Other information:




### FOR PR REVIEWER ONLY:
